projects/packages/my-jetpack/src/class-rest-recommendations-evaluation.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * Sets up the Evaluation Recommendations REST API endpoints.
+ *
+ * @package automattic/my-jetpack
+ */
+
+namespace Automattic\Jetpack\My_Jetpack;
+
+use Automattic\Jetpack\Connection\Client;
+use Automattic\Jetpack\Connection\Manager as Connection_Manager;
+use WP_Error;
+
+/**
+ * Registers the REST routes for Evaluation Recommendations.
+ */
+class REST_Recommendations_Evaluation {
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		register_rest_route(
+			'my-jetpack/v1',
+			'/site/recommendations/evaluation/',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => __CLASS__ . '::evaluate_site_recommendations',
+					'permission_callback' => __CLASS__ . '::permissions_callback',
+				),
+			)
+		);
+
+		register_rest_route(
+			'my-jetpack/v1',
+			'/site/recommendations/save-evaluation/',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::EDITABLE,
+					'callback'            => __CLASS__ . '::save_evaluation_recommendations',
+					'permission_callback' => __CLASS__ . '::permissions_callback',
+				),
+			)
+		);
+	}
+
+	/**
+	 * Check user capability to access the endpoint.
+	 *
+	 * @access public
+	 * @static
+	 *
+	 * @return true|WP_Error
+	 */
+	public static function permissions_callback() {
+		$connection        = new Connection_Manager();
+		$is_site_connected = $connection->is_connected();
+
+		if ( ! $is_site_connected ) {
+			return new WP_Error(
+				'not_connected',
+				__( 'Your site is not connected to Jetpack.', 'jetpack-my-jetpack' ),
+				array(
+					'status' => 400,
+				)
+			);
+		}
+
+		return true; // We require site to be connected.
+	}
+
+	/**
+	 * Recommendations Evaluation endpoint.
+	 *
+	 * @param object $request Query request.
+	 *
+	 * @return array of 3 product slugs (recommendations).
+	 */
+	public static function evaluate_site_recommendations( $request ) {
+		$goals = $request->get_param( 'goals' );
+
+		if ( ! isset( $goals ) ) {
+			return new WP_Error( 'missing_goals', 'Goals are required', array( 'status' => 400 ) );
```

Comment:
Looks like we have some type issues here and below

Reply:
yeah 😬 will fix that up